### PR TITLE
ADTS-113

### PR DESCRIPTION
**Pull request by JoseZarmada**
### Testing 🧪
 > [ Add any special instructions needed to test or view your changes such as environment URLs or other config details. ]

### Checklist ✔️
- [ ] Linked associated issue (if present)
- [ ] Added relevant labels
- [ ] Requested developer reviews
- [ ] Request UI review from design / PM
- [ ] Verify all code checks are passing

### CODE REVIEW

File: src/Components/OATHeader/OATHeader.tsx [hunk 1]
```diff
@@ -77,6 +91,42 @@ const OATHeader = ({ elements, onImportClick }: OATHeaderProps) => {
...
+                try {
+                    const content = await current.text();
+                    newItem.content = JSON.parse(content);
+                } catch (error) {
+                    console.log(Error(error));
+                    alert(Error(error));
+                }
+                items.push(newItem.content);
+            }
+            dispatch;
```
Comment (jamescarpinter) [inline src/Components/OATHeader/OATHeader.tsx:122]: What is this line?

File: src/Components/OATHeader/OATHeader.tsx [hunk 2]
```diff
@@ -77,6 +91,42 @@ const OATHeader = ({ elements, onImportClick }: OATHeaderProps) => {
...
+                const newItem = {
+                    name: current.name,
+                    size: prettyBytes(current.size),
+                    status: FileUploadStatus.Uploading
+                } as IFileItem;
+                try {
+                    const content = await current.text();
+                    newItem.content = JSON.parse(content);
+                } catch (error) {
+                    console.log(Error(error));
```
Comment (jamescarpinter) [inline src/Components/OATHeader/OATHeader.tsx:117]: Why do we need to wrap this is an error?

File: src/Components/OATHeader/OATHeader.tsx [hunk 3]
```diff
@@ -77,6 +91,42 @@ const OATHeader = ({ elements, onImportClick }: OATHeaderProps) => {
         }
     ];
 
+    useEffect(() => {
+        const newFiles = [];
+        acceptedFiles.forEach((sF) => {
+            if (sF.type === 'application/json') {
+                newFiles.push(sF);
+            }
```
Comment (jamescarpinter) [inline src/Components/OATHeader/OATHeader.tsx:99]: How do we inform the user about files that we don't accept?

File: src/Components/OATGraphViewer/OATGraphViewer.tsx [hunk 4]
```diff
@@ -131,85 +131,100 @@ const OATGraphViewer = ({ state, dispatch }: OATGraphProps) => {
...
                     (element) => element.id === input['@id']
                 );
-                if (!node) {
-                    let relationships = [];
-                    let contents = [];
-                    input['contents'].forEach((content) => {
-                        if (content['@type'] === OATComponentHandleName) {
-                            const componentRelationship = new ElementEdge(
+                if (node) {
+                    importModelsList = importModelsList.reduce(
```
Comment (jamescarpinter) [inline src/Components/OATGraphViewer/OATGraphViewer.tsx:141]: Can you help me understand what's happening in this reduce function? If we're filtering, then `filter` might be a better option?

**Review by jamescarpinter — CHANGES_REQUESTED** [review]
Initial comment inline. Thanks.

File: src/Components/OATHeader/OATHeader.tsx (hunk 1 above)
Comment (JoseZarmada) [reply src/Components/OATHeader/OATHeader.tsx:122]: Typo Just Fixed

File: src/Components/OATHeader/OATHeader.tsx (hunk 2 above)
Comment (JoseZarmada) [reply src/Components/OATHeader/OATHeader.tsx:117]: Took that from the example class, and Milton is actually going to be working In the Error Handling Component, so we should be using that instead eventually

File: src/Components/OATHeader/OATHeader.tsx (hunk 3 above)
Comment (JoseZarmada) [reply src/Components/OATHeader/OATHeader.tsx:99]: Then Again using The Error Handling Component should be a good Idea for this Also

Comment (jamescarpinter) [reply src/Components/OATHeader/OATHeader.tsx:99]: Let's use an alert until then.

File: src/Components/OATHeader/OATHeader.tsx (hunk 2 above)
Comment (jamescarpinter) [reply src/Components/OATHeader/OATHeader.tsx:117]: Can you please provide a link to the example class?

I would've expected a read/parse error to result in `error` already being of type `Error`. I.e. can't we just `console.log(error);`?

**Review by jamescarpinter — APPROVED** [review]

**Pull request merged**